File: DotNetGotchas/VBDN/NotToFinalize/Modified1/FinalizeAndManagedResources/SomeOtherClass.vb
'SomeOtherClass.vb
Public Class SomeOtherClass
    Private values() As Integer = New Integer(1000000 - 1) {}

    Public Sub Notify()
        Console.WriteLine("Entering Notify")
        SyncLock Me
            Console.WriteLine("Notify called on SomeOtherClass")
        End SyncLock
    End Sub


    Protected Overrides Sub Finalize()
        MyBase.Finalize()
        Console.WriteLine("*** SomeOtherClass Finalized")
    End Sub
End Class

File: DotNetGotchas/CSharp/NotToFinalize/Modified1/FinalizeAndManagedResources/SomeOtherClass.cs
//SomeOtherClass.cs
using System;

namespace FinalizeAndManagedResources
{
	public class SomeOtherClass
	{
		private int[] values = new int[1000000];

		public void Notify()
		{
			Console.WriteLine("Entering Notify");
			lock(this)
			{
				Console.WriteLine("Notify called on SomeOtherClass");
			}
		}

		~SomeOtherClass()
		{
			Console.WriteLine("*** SomeOtherClass Finalized");
		}
	}
}
